### PR TITLE
api: Rename *SortConstructor* to *UninterpretedSortConstructor*.

### PR DESCRIPTION
**Pull request by aniemetz**

### CODE REVIEW

File: src/api/cpp/cvc5.h [hunk 1]
```diff
@@ -584,19 +584,25 @@ class CVC5_EXPORT Sort
   bool isUninterpretedSort() const;
 
   /**
-   * Is this a sort constructor kind?
+   * Is this an uninterpreted sort constructor kind?
+   *
+   * An uninterpreted sort constructor is an uninterpreted sort with arity > 0.
```
Comment (ajreynol) [inline src/api/cpp/cvc5.h:589]: I would not call it an uninterpreted sort.  Note that calling instantiate on it generates sorts that are essentially uninterpreted sorts.

This should perhaps say that it was constructed via `declareSort` with arity > 0.

File: src/api/java/io/github/cvc5/api/Solver.java [hunk 2]
```diff
@@ -1555,6 +1558,8 @@ private native long declareFun(
    * {@code
    *   ( declare-sort <symbol> <numeral> )
    * }
+   * @apiNote This corresponds to mkUninterpretedSort() if arity = 0, and to
+   *          mkUninterpretedSortConstructorSort() if arity > 0.
```
Comment (ajreynol) [inline src/api/java/io/github/cvc5/api/Solver.java:1562]:
```suggestion
   *          mkUninterpretedSortConstructorSort(arity) if arity > 0.
```

File: src/api/cpp/cvc5.h [hunk 3]
```diff
@@ -3841,6 +3851,9 @@ class CVC5_EXPORT Solver
    *     (declare-sort <symbol> <numeral>)
    * \endverbatim
    *
+   * @note This corresponds to mkUninterpretedSort() if arity = 0, and to
+   *       mkUninterpretedSortConstructorSort() if arity > 0.
```
Comment (ajreynol) [inline src/api/cpp/cvc5.h:3855]:
```suggestion
   *       mkUninterpretedSortConstructorSort(arity) if arity > 0.
```

File: src/api/python/cvc5.pxi [hunk 4]
```diff
@@ -1842,6 +1846,10 @@ cdef class Solver:
 
             ( declare-sort <symbol> <numeral> )
 
+        .. note::
+          This corresponds to mkUninterpretedSort() if arity = 0, and to
+          mkUninterpretedSortConstructorSort() if arity > 0.
```
Comment (ajreynol) [inline src/api/python/cvc5.pxi:1851]:
```suggestion
          mkUninterpretedSortConstructorSort(arity) if arity > 0.
```

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: src/api/python/cvc5.pxi (hunk 4 above)
Comment (aniemetz) [reply src/api/python/cvc5.pxi:1851]: Cannot be cross-referenced when arguments are given, won't fix.

File: src/api/java/io/github/cvc5/api/Solver.java (hunk 2 above)
Comment (aniemetz) [reply src/api/java/io/github/cvc5/api/Solver.java]: We can't use the types of the arguments in the java doc, so I'll leave it as it is and have only fixed the problem with `>`.

**Review by ajreynol — APPROVED** [review]

**Pull request merged**